### ISSUE TITLE
IndexError: list index out of range

### ISSUE THREAD

**Issue by notflip** [CyberPunkMetalHead/Binance-volatility-trading-bot#9]
```
 ETHUSDT has gained 21.425% in the last 5 minutes, calculating volume in USDT
 preparing to buy 0.26192 ETHUSDT
Traceback (most recent call last):
  File "/Users/miguelstevens/Desktop/bot/Binance Detect Moonings.py", line 299, in <module>
    update_porfolio(orders, last_price, volume)
  File "/Users/miguelstevens/Desktop/bot/Binance Detect Moonings.py", line 267, in update_porfolio
    'symbol': orders[coin][0]['symbol'],
IndexError: list index out of range
```

Received this error, the console has been open for about an hour, all default settings are there, using testnet as well.

**Comment by CyberPunkMetalHead** [comment]
I'm unable to replicate your issue but could you try changing the way you store information from the last order in the update portfiolio function to be:

```
            'symbol': orders[coin][-1]['symbol'],
            'orderid': orders[coin][-1]['orderId'],
            'timestamp': orders[coin][-1]['time'],
            'bought_at': last_price[coin]['price'],
            'volume': volume[coin]
```

Let me know if this fixes your issue

**Comment by notflip** [comment]
@CyberPunkMetalHead I made the changes! I'll let you know if the problem is fixed with that, thank you!

**Comment by CyberPunkMetalHead** [comment]
Awesome, keep me updated :)

**Comment by notflip** [comment]
Just out of curiosity,  if 0 selects the first index of the array, what does -1 do in python, select the last?

**Comment by CyberPunkMetalHead** [comment]
Yep, exactly that :)

**Comment by goodlifeinc** [comment]
Hi, I am experiencing the same error.
It appears get_all_orders doesn't return the active order since it has already been settled?

`
2021-05-09 23:33:39: ATMUSDT has gained 4.953% in the last 5 minutes, calculating volume in USDT                                                                        2021-05-09 23:33:39:  preparing to buy 1.443 ATMUSDT                                                                                                                    {                                                                                                                                                                           "ATMUSDT": []                                                                                                                                                       }                                                                                                                                                                       Traceback (most recent call last):                                                                                                                                        File "...\Binance Detect Moonings.py", line 304, in <module>                                                    update_porfolio(orders, last_price, volume)                                                                                                                           File "...\Binance Detect Moonings.py", line 272, in update_porfolio                                             'symbol': orders[coin][-1]['symbol'],                                                                                                                               IndexError: list index out of range
`

**Comment by CyberPunkMetalHead** [comment]
I believe I have found a fix for this as well as the cause of the issue. Binance seems to take a while (sometimes up to 3 seconds until it returns the results of a live order). I have added a while block that will keep on calling the API once every second it returns the order. The output will look something like this

`Binance is being slow in returning the order, again the API again...
Binance is being slow in returning the order, again the API again...
Binance is being slow in returning the order, again the API again...
Order returned, saving order to file
 preparing to buy 110.0 DOCKUSDT`

I will shortly commit the changes to the main branch
